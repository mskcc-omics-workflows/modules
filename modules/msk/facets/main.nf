process FACETS {
    tag "$meta.id"
    label 'process_medium'

    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'docker://mskcc/facets-suite:2.0.9':
        'docker.io/mskcc/facets-suite:2.0.9' }"


    input:

    tuple val(meta), path(snp_pileup), val(legacy_output_mode)  //  [ meta, ${prefix}.snp_pileup.gz, false]


    output:
    tuple val(meta), path("*_purity.seg")              , emit: purity_seg
    tuple val(meta), path("*_purity.?d*")              , emit: purity_r_data
    tuple val(meta), path("*_purity*png")              , emit: purity_png
    tuple val(meta), path("*_purity.out")              , emit: purity_out, optional: true
    tuple val(meta), path("*_purity.cncf.txt")         , emit: purity_cncf_txt, optional: true
    tuple val(meta), path("*_hisens.seg")              , emit: hisens_seg
    tuple val(meta), path("*_hisens.?d*")              , emit: hisens_r_data
    tuple val(meta), path("*_hisens*png")              , emit: hisens_png
    tuple val(meta), path("*_hisens.out")              , emit: hisens_out, optional: true
    tuple val(meta), path("*_hisens.cncf.txt")         , emit: hisens_cncf_txt, optional: true
    tuple val(meta), path("*.qc.txt")                  , emit: qc_txt
    tuple val(meta), path("*.gene_level.txt")          , emit: gene_level_txt
    tuple val(meta), path("*.arm_level.txt")           , emit: arm_level_txt
    tuple val(meta), path("${meta.id}.txt")            , emit: output_txt, optional: true
    path "versions.yml"                                , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def legacy_output_arg = legacy_output_mode ? '--legacy-output TRUE' : ''

    """
    /usr/bin/facets-suite/run-facets-wrapper.R \
        ${args} \
        ${legacy_output_arg} \
        --sample-id ${prefix} \
        --counts-file ${snp_pileup}
    
    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        facets_suite: \$(Rscript -e "packageVersion('facetsSuite')" | grep -oP "\\d+.\\d+.\\d+")
        facets: \$(Rscript -e "packageVersion('facets')" | grep -oP "\\d+.\\d+.\\d+")
        r: \$(R --version | grep -oP '(?<=R version ).*(?=\\()')
        pctGCdata: \$(Rscript -e "packageVersion('pctGCdata')" | grep -oP "\\d+.\\d+.\\d+")
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}_purity.seg
    touch ${prefix}_purity.CNCF.png
    touch ${prefix}_purity.png
    touch ${prefix}_purity.cncf.txt
    touch ${prefix}_purity.Rdata
    touch ${prefix}_purity.rds
    touch ${prefix}_purity.out
    touch ${prefix}_hisens.seg
    touch ${prefix}_hisens.CNCF.png
    touch ${prefix}_hisens.png
    touch ${prefix}_hisens.cncf.txt
    touch ${prefix}_hisens.Rdata
    touch ${prefix}_hisens.rds
    touch ${prefix}_hisens.out
    touch ${prefix}.qc.txt
    touch ${prefix}.gene_level.txt
    touch ${prefix}.arm_level.txt
    touch ${prefix}.txt



    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        facets_suite: \$(Rscript -e "packageVersion('facetsSuite')" | grep -oP "\\d+.\\d+.\\d+")
        facets: \$(Rscript -e "packageVersion('facets')" | grep -oP "\\d+.\\d+.\\d+")
        r: \$(R --version | grep -oP '(?<=R version ).*(?=\\()')
        pctGCdata: \$(Rscript -e "packageVersion('pctGCdata')" | grep -oP "\\d+.\\d+.\\d+")
    END_VERSIONS
    """
}