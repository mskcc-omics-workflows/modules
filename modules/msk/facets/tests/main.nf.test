nextflow_process {

    name "Test FACETS"
    script "../main.nf"
    process "FACETS"
    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "modules_msk"
    tag "facets"
    tag "snppileup"

    test("facets - pileup - legacy output") {

        setup {
            run("SNPPILEUP"){
                script "../../snppileup/main.nf"
                process {
                    """                
                    meta = [ id:'test' ] // meta map
                    input[0] = [
                                    meta,
                                    file(params.test_data_mskcc['bam_files']['test_argos_sample_1_normal_bam'], checkIfExists: true),
                                    file(params.test_data_mskcc['bam_files']['test_argos_sample_1_normal_bam_bai'], checkIfExists: true),
                                    file(params.test_data_mskcc['bam_files']['test_argos_sample_1_tumor_bam'], checkIfExists: true),
                                    file(params.test_data_mskcc['bam_files']['test_argos_sample_1_tumor_bam_bai'], checkIfExists: true),
                                    [],
                                    []
                                ]
                    input[1] = [
                                    meta,
                                    file(params.test_data_mskcc['hg37']['dbsnp_138_b37_chr17_vcf_gz'], checkIfExists: true),
                                    file(params.test_data_mskcc['hg37']['dbsnp_138_b37_chr17_vcf_gz_tbi'], checkIfExists: true)
                               ]
                    """
                }

            }
        }

        when {
            process {
                """                
                meta = [ id:'test' ] // meta map
                input[0] = SNPPILEUP.out.pileup.collect{ meta, pileup -> pileup }.map{ pileup -> [meta, pileup, true]}               

                """
                                
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("facets - pileup - output") {

        setup {
            run("SNPPILEUP"){
                script "../../snppileup/main.nf"
                process {
                    """                
                    meta = [ id:'test' ] // meta map
                    input[0] = [
                                    meta,
                                    file(params.test_data_mskcc['bam_files']['test_argos_sample_1_normal_bam'], checkIfExists: true),
                                    file(params.test_data_mskcc['bam_files']['test_argos_sample_1_normal_bam_bai'], checkIfExists: true),
                                    file(params.test_data_mskcc['bam_files']['test_argos_sample_1_tumor_bam'], checkIfExists: true),
                                    file(params.test_data_mskcc['bam_files']['test_argos_sample_1_tumor_bam_bai'], checkIfExists: true),
                                    [],
                                    []
                                ]
                    input[1] = [
                                    meta,
                                    file(params.test_data_mskcc['hg37']['dbsnp_138_b37_chr17_vcf_gz'], checkIfExists: true),
                                    file(params.test_data_mskcc['hg37']['dbsnp_138_b37_chr17_vcf_gz_tbi'], checkIfExists: true)
                               ]
                    """
                }

            }
        }

        when {
            process {
                """                
                meta = [ id:'test' ] // meta map
                input[0] = SNPPILEUP.out.pileup.collect{ meta, pileup -> pileup }.map{ pileup -> [meta, pileup, false]}               

                """
                                
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("facets - pileup - legacy output stub") {

        options "-stub"

        when {
            process {
                """
                    meta = [ id:'test' ] // meta map
                    input[0] = [
                                    meta,
                                    file("stub_file"),
                                    true
                                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("facets - pileup - output stub") {

        options "-stub"

        when {
            process {
                """
                    meta = [ id:'test' ] // meta map
                    input[0] = [
                                    meta,
                                    file("stub_file"),
                                    false
                                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
