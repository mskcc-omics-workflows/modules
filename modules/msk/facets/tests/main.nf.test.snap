{
    "facets - pileup - output": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_purity.seg:md5,a4e7cf69e12b7d7e2d1ccec81cd1c032"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test_purity.rds:md5,4fcba441dd22120332fc490c118c2457"
                    ]
                ],
                "10": [
                    [
                        {
                            "id": "test"
                        },
                        "test.qc.txt:md5,97f248167bc6d61d714a1a8fb49b077f"
                    ]
                ],
                "11": [
                    [
                        {
                            "id": "test"
                        },
                        "test.gene_level.txt:md5,4b94d595242700da7ea15d362b4e498c"
                    ]
                ],
                "12": [
                    [
                        {
                            "id": "test"
                        },
                        "test.arm_level.txt:md5,06846b84c731a5768f4cc996de2098df"
                    ]
                ],
                "13": [
                    [
                        {
                            "id": "test"
                        },
                        "test.txt:md5,c526463b32c1a287d851acb79cf1c790"
                    ]
                ],
                "14": [
                    "versions.yml:md5,d89b44510220c64887a3b4b0112c182b"
                ],
                "2": [
                    [
                        {
                            "id": "test"
                        },
                        "test_purity.png:md5,569501072d3e0e51c57f35993b985e8e"
                    ]
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "5": [
                    [
                        {
                            "id": "test"
                        },
                        "test_hisens.seg:md5,a4e7cf69e12b7d7e2d1ccec81cd1c032"
                    ]
                ],
                "6": [
                    [
                        {
                            "id": "test"
                        },
                        "test_hisens.rds:md5,801a9844163f5c921e415cee8981d23e"
                    ]
                ],
                "7": [
                    [
                        {
                            "id": "test"
                        },
                        "test_hisens.png:md5,17a0c2fac69e0fff252ed2f5b87d856b"
                    ]
                ],
                "8": [
                    
                ],
                "9": [
                    
                ],
                "arm_level_txt": [
                    [
                        {
                            "id": "test"
                        },
                        "test.arm_level.txt:md5,06846b84c731a5768f4cc996de2098df"
                    ]
                ],
                "gene_level_txt": [
                    [
                        {
                            "id": "test"
                        },
                        "test.gene_level.txt:md5,4b94d595242700da7ea15d362b4e498c"
                    ]
                ],
                "hisens_cncf_txt": [
                    
                ],
                "hisens_out": [
                    
                ],
                "hisens_png": [
                    [
                        {
                            "id": "test"
                        },
                        "test_hisens.png:md5,17a0c2fac69e0fff252ed2f5b87d856b"
                    ]
                ],
                "hisens_r_data": [
                    [
                        {
                            "id": "test"
                        },
                        "test_hisens.rds:md5,801a9844163f5c921e415cee8981d23e"
                    ]
                ],
                "hisens_seg": [
                    [
                        {
                            "id": "test"
                        },
                        "test_hisens.seg:md5,a4e7cf69e12b7d7e2d1ccec81cd1c032"
                    ]
                ],
                "output_txt": [
                    [
                        {
                            "id": "test"
                        },
                        "test.txt:md5,c526463b32c1a287d851acb79cf1c790"
                    ]
                ],
                "purity_cncf_txt": [
                    
                ],
                "purity_out": [
                    
                ],
                "purity_png": [
                    [
                        {
                            "id": "test"
                        },
                        "test_purity.png:md5,569501072d3e0e51c57f35993b985e8e"
                    ]
                ],
                "purity_r_data": [
                    [
                        {
                            "id": "test"
                        },
                        "test_purity.rds:md5,4fcba441dd22120332fc490c118c2457"
                    ]
                ],
                "purity_seg": [
                    [
                        {
                            "id": "test"
                        },
                        "test_purity.seg:md5,a4e7cf69e12b7d7e2d1ccec81cd1c032"
                    ]
                ],
                "qc_txt": [
                    [
                        {
                            "id": "test"
                        },
                        "test.qc.txt:md5,97f248167bc6d61d714a1a8fb49b077f"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d89b44510220c64887a3b4b0112c182b"
                ]
            }
        ],
        "timestamp": "2024-02-27T11:00:33.950903"
    },
    "facets - pileup - output stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_purity.seg:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "test_purity.Rdata:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_purity.rds:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "10": [
                    [
                        {
                            "id": "test"
                        },
                        "test.qc.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "11": [
                    [
                        {
                            "id": "test"
                        },
                        "test.gene_level.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "12": [
                    [
                        {
                            "id": "test"
                        },
                        "test.arm_level.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "13": [
                    [
                        {
                            "id": "test"
                        },
                        "test.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "14": [
                    "versions.yml:md5,d89b44510220c64887a3b4b0112c182b"
                ],
                "2": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "test_purity.CNCF.png:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_purity.png:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test"
                        },
                        "test_purity.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "test"
                        },
                        "test_purity.cncf.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "5": [
                    [
                        {
                            "id": "test"
                        },
                        "test_hisens.seg:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "6": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "test_hisens.Rdata:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_hisens.rds:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "7": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "test_hisens.CNCF.png:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_hisens.png:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "8": [
                    [
                        {
                            "id": "test"
                        },
                        "test_hisens.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "9": [
                    [
                        {
                            "id": "test"
                        },
                        "test_hisens.cncf.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "arm_level_txt": [
                    [
                        {
                            "id": "test"
                        },
                        "test.arm_level.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "gene_level_txt": [
                    [
                        {
                            "id": "test"
                        },
                        "test.gene_level.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "hisens_cncf_txt": [
                    [
                        {
                            "id": "test"
                        },
                        "test_hisens.cncf.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "hisens_out": [
                    [
                        {
                            "id": "test"
                        },
                        "test_hisens.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "hisens_png": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "test_hisens.CNCF.png:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_hisens.png:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "hisens_r_data": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "test_hisens.Rdata:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_hisens.rds:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "hisens_seg": [
                    [
                        {
                            "id": "test"
                        },
                        "test_hisens.seg:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "output_txt": [
                    [
                        {
                            "id": "test"
                        },
                        "test.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "purity_cncf_txt": [
                    [
                        {
                            "id": "test"
                        },
                        "test_purity.cncf.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "purity_out": [
                    [
                        {
                            "id": "test"
                        },
                        "test_purity.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "purity_png": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "test_purity.CNCF.png:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_purity.png:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "purity_r_data": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "test_purity.Rdata:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_purity.rds:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "purity_seg": [
                    [
                        {
                            "id": "test"
                        },
                        "test_purity.seg:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "qc_txt": [
                    [
                        {
                            "id": "test"
                        },
                        "test.qc.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d89b44510220c64887a3b4b0112c182b"
                ]
            }
        ],
        "timestamp": "2024-02-27T11:00:50.172557"
    },
    "facets - pileup - legacy output": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_purity.seg:md5,a4e7cf69e12b7d7e2d1ccec81cd1c032"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test_purity.Rdata:md5,e28f0252fb36358fb3c70920fb402033"
                    ]
                ],
                "10": [
                    [
                        {
                            "id": "test"
                        },
                        "test.qc.txt:md5,97f248167bc6d61d714a1a8fb49b077f"
                    ]
                ],
                "11": [
                    [
                        {
                            "id": "test"
                        },
                        "test.gene_level.txt:md5,4b94d595242700da7ea15d362b4e498c"
                    ]
                ],
                "12": [
                    [
                        {
                            "id": "test"
                        },
                        "test.arm_level.txt:md5,06846b84c731a5768f4cc996de2098df"
                    ]
                ],
                "13": [
                    
                ],
                "14": [
                    "versions.yml:md5,d89b44510220c64887a3b4b0112c182b"
                ],
                "2": [
                    [
                        {
                            "id": "test"
                        },
                        "test_purity.CNCF.png:md5,569501072d3e0e51c57f35993b985e8e"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test"
                        },
                        "test_purity.out:md5,8a93be2bafdcc20ad71ba660b19157d6"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "test"
                        },
                        "test_purity.cncf.txt:md5,4083bfc9655c5b4ff48a8d9bd2aa69d3"
                    ]
                ],
                "5": [
                    [
                        {
                            "id": "test"
                        },
                        "test_hisens.seg:md5,a4e7cf69e12b7d7e2d1ccec81cd1c032"
                    ]
                ],
                "6": [
                    [
                        {
                            "id": "test"
                        },
                        "test_hisens.Rdata:md5,9bc324b7329a148a805647ded5d2f113"
                    ]
                ],
                "7": [
                    [
                        {
                            "id": "test"
                        },
                        "test_hisens.CNCF.png:md5,17a0c2fac69e0fff252ed2f5b87d856b"
                    ]
                ],
                "8": [
                    [
                        {
                            "id": "test"
                        },
                        "test_hisens.out:md5,2a51d7227f21a4d0d47ccc459200a279"
                    ]
                ],
                "9": [
                    [
                        {
                            "id": "test"
                        },
                        "test_hisens.cncf.txt:md5,dabe379f3f9acc9a489879678eb297d8"
                    ]
                ],
                "arm_level_txt": [
                    [
                        {
                            "id": "test"
                        },
                        "test.arm_level.txt:md5,06846b84c731a5768f4cc996de2098df"
                    ]
                ],
                "gene_level_txt": [
                    [
                        {
                            "id": "test"
                        },
                        "test.gene_level.txt:md5,4b94d595242700da7ea15d362b4e498c"
                    ]
                ],
                "hisens_cncf_txt": [
                    [
                        {
                            "id": "test"
                        },
                        "test_hisens.cncf.txt:md5,dabe379f3f9acc9a489879678eb297d8"
                    ]
                ],
                "hisens_out": [
                    [
                        {
                            "id": "test"
                        },
                        "test_hisens.out:md5,2a51d7227f21a4d0d47ccc459200a279"
                    ]
                ],
                "hisens_png": [
                    [
                        {
                            "id": "test"
                        },
                        "test_hisens.CNCF.png:md5,17a0c2fac69e0fff252ed2f5b87d856b"
                    ]
                ],
                "hisens_r_data": [
                    [
                        {
                            "id": "test"
                        },
                        "test_hisens.Rdata:md5,9bc324b7329a148a805647ded5d2f113"
                    ]
                ],
                "hisens_seg": [
                    [
                        {
                            "id": "test"
                        },
                        "test_hisens.seg:md5,a4e7cf69e12b7d7e2d1ccec81cd1c032"
                    ]
                ],
                "output_txt": [
                    
                ],
                "purity_cncf_txt": [
                    [
                        {
                            "id": "test"
                        },
                        "test_purity.cncf.txt:md5,4083bfc9655c5b4ff48a8d9bd2aa69d3"
                    ]
                ],
                "purity_out": [
                    [
                        {
                            "id": "test"
                        },
                        "test_purity.out:md5,8a93be2bafdcc20ad71ba660b19157d6"
                    ]
                ],
                "purity_png": [
                    [
                        {
                            "id": "test"
                        },
                        "test_purity.CNCF.png:md5,569501072d3e0e51c57f35993b985e8e"
                    ]
                ],
                "purity_r_data": [
                    [
                        {
                            "id": "test"
                        },
                        "test_purity.Rdata:md5,e28f0252fb36358fb3c70920fb402033"
                    ]
                ],
                "purity_seg": [
                    [
                        {
                            "id": "test"
                        },
                        "test_purity.seg:md5,a4e7cf69e12b7d7e2d1ccec81cd1c032"
                    ]
                ],
                "qc_txt": [
                    [
                        {
                            "id": "test"
                        },
                        "test.qc.txt:md5,97f248167bc6d61d714a1a8fb49b077f"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d89b44510220c64887a3b4b0112c182b"
                ]
            }
        ],
        "timestamp": "2024-02-27T10:59:59.751739"
    },
    "facets - pileup - legacy output stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_purity.seg:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "test_purity.Rdata:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_purity.rds:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "10": [
                    [
                        {
                            "id": "test"
                        },
                        "test.qc.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "11": [
                    [
                        {
                            "id": "test"
                        },
                        "test.gene_level.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "12": [
                    [
                        {
                            "id": "test"
                        },
                        "test.arm_level.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "13": [
                    [
                        {
                            "id": "test"
                        },
                        "test.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "14": [
                    "versions.yml:md5,d89b44510220c64887a3b4b0112c182b"
                ],
                "2": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "test_purity.CNCF.png:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_purity.png:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test"
                        },
                        "test_purity.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "test"
                        },
                        "test_purity.cncf.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "5": [
                    [
                        {
                            "id": "test"
                        },
                        "test_hisens.seg:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "6": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "test_hisens.Rdata:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_hisens.rds:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "7": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "test_hisens.CNCF.png:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_hisens.png:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "8": [
                    [
                        {
                            "id": "test"
                        },
                        "test_hisens.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "9": [
                    [
                        {
                            "id": "test"
                        },
                        "test_hisens.cncf.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "arm_level_txt": [
                    [
                        {
                            "id": "test"
                        },
                        "test.arm_level.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "gene_level_txt": [
                    [
                        {
                            "id": "test"
                        },
                        "test.gene_level.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "hisens_cncf_txt": [
                    [
                        {
                            "id": "test"
                        },
                        "test_hisens.cncf.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "hisens_out": [
                    [
                        {
                            "id": "test"
                        },
                        "test_hisens.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "hisens_png": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "test_hisens.CNCF.png:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_hisens.png:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "hisens_r_data": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "test_hisens.Rdata:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_hisens.rds:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "hisens_seg": [
                    [
                        {
                            "id": "test"
                        },
                        "test_hisens.seg:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "output_txt": [
                    [
                        {
                            "id": "test"
                        },
                        "test.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "purity_cncf_txt": [
                    [
                        {
                            "id": "test"
                        },
                        "test_purity.cncf.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "purity_out": [
                    [
                        {
                            "id": "test"
                        },
                        "test_purity.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "purity_png": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "test_purity.CNCF.png:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_purity.png:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "purity_r_data": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "test_purity.Rdata:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_purity.rds:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "purity_seg": [
                    [
                        {
                            "id": "test"
                        },
                        "test_purity.seg:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "qc_txt": [
                    [
                        {
                            "id": "test"
                        },
                        "test.qc.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d89b44510220c64887a3b4b0112c182b"
                ]
            }
        ],
        "timestamp": "2024-02-27T11:00:41.714449"
    }
}